en/docs/extended_topics/python_kernel_op.md

```diff
@@ -0,0 +1,291 @@
+    output = MyJob(input)
+    print(input)
+    print(output)
+```
+
+In the above code, the module object is created by `flow.experimental.custom_op_module` which takes two parameters. The first one is `op_type_name` and the second one is the path to the `user_relu`. The returned `module` object represents our custom Op.
+
+Next, the custom Op can be enabled by `user_sigmoid_op.py_api().cpp_def().py_kernel().build_load().` The Python interface of Op when it is enabled is defined in `user_relu_py_api.py' as (`user_relu_forward`). It is placed in the `api` namespace of the `moudle` object. Therefore, we need to call it in the following way:
+
+```python
+user_sigmoid_op.api.user_relu_forward(x)
+```
+
+Because the Python Kernel can only run on CPU devices which means you need to specify the calculation device as CPU:
+```python
+with flow.scope.placement("cpu", "0:0"):
+```
+
+## Provides Inverse Calculation for Custom Op
+We have already done the forward calculation of the `user_relu` in the above steps which can be used for the job function of `type="predict"` . However, if we want to support a training job function of `type="train"`, we need to provide the reverse calculation for the custom Op as well.
+
+The code that provides the reverse calculation for the custom Op needs to be written in the `op_type_name_cpp_def.cpp` and registered with the macro `REGISTER_USER_OP_GRAD`.
+
+From a mathematical point of view, the registration process is where we specify a backward gradient calculation method for a custom op. From a programming point of view, it is setting up a backward-generating function for a custom Op. Within that function, writing code that specifies how the input gradient of the Op is calculated.
+
+In the following, we will implement a special Op called `user_relu_backward`. We will use this "specially tailored" Op when registering a backward gradient for `user_relu`.
+
+### Implementation of `user_relu_backward` Op
+The process for implementing the `user_relu_backward` is almost identical to implementing the `user_relu`. First, configure and register the Op in `user_relu_cpp_def.cpp`:
+
+```cpp
+REGISTER_USER_OP("user_relu_backward")
+    .Input("y")
+    .Input("dy")
+    .Output("dx")
+    .Attr<std::string>("device_sub_tag", "py")
+    .SetTensorDescInferFn([](user_op::InferContext* ctx) -> Maybe<void> {
+      const Shape* dy_shape = ctx->Shape4ArgNameAndIndex("dy", 0);
+      Shape* dx_shape = ctx->Shape4ArgNameAndIndex("dx", 0);
+      *dx_shape = *dy_shape;
+      return Maybe<void>::Ok();
+    });
+```
+
+Please pay attention that`.Attr<std::string>("device_sub_tag", "py")` must in the above code. It tells OneFlow to call the Python Kernel by default when using this Op.
+
+By the same logic, since we don't need to call `user_relu_backward`. So you don't need to wrap the Python interface to `user_relu_py_api.py` for `user_relu_backward`.  We can implement it directly in the Python Kernel.
```

Comment:
```suggestion
By the same logic, since the users don't need to call `user_relu_backward` directly. So we don't need to provide the Python wrapper for `user_relu_backward` in `user_relu_py_api.py`.  We just need to implement its Python Kernel.
```